### ISSUE TITLE
Allow JAX backend in QuantumModel

### ISSUE THREAD

**Issue by dominikandreasseitz** [pasqal-io/qadence#203]

**Comment by jpmoutinho** [comment]
As discussed with @dominikandreasseitz this requires a bit of a rethinking of how the `QuantumModel` is implemented in Qadence.

Changing the `QuantumModel` from a `torch.nn.Module` could also cause compatibility issues with torch CUDA.

**Comment by jpmoutinho** [comment]
@Roland-djee @dominikandreasseitz I would say this is not a priority for the current QuantumModel

**Comment by Roland-djee** [comment]
> @Roland-djee @dominikandreasseitz I would say this is not a priority for the current QuantumModel

Agreed. There is not enough push in that direction company-wise. Though this should be resolved in the next gen.
